### PR TITLE
TPR migration guidelines

### LINKED ISSUE #819: Document how to migrate existing TPR's to CRD's
Version 1.7.1 removed support for TPR's. Rook needs to migrate any existing TPR's on a cluster to CRD's.

### PR DESCRIPTION
**Pull request by paha**
resolves #819
[skip ci]

### CODE REVIEW

File: Documentation/upgrade.md [hunk 1]
```diff
@@ -20,6 +20,10 @@ With this manual upgrade guide, there are a few notes to consider:
...
 * Rook is still in an alpha state.  Migrations and general support for breaking changes across versions are not supported or covered in this guide.
 * This guide assumes that your Rook operator and its agents are running in the `rook-system` namespace. It also assumes that your Rook cluster is in the `rook` namespace.  If any of these components is in a different namespace, search/replace all instances of `-n rook-system` and `-n rook` in this guide with `-n <your namespace>`.
+* Rook cluster installations on Kubernetes prior to version 1.7.x, use [ThirdPartyResource](https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-third-party-resource/) that have been deprecated as of 1.7 and removed in 1.8. When upgrading your Kubernetes cluster Rook TPRs have to be migrated to CustomResourceDefinition (CRD) following [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/migrate-third-party-resource/). Rook TPRs that require migration during upgrade are: 
+  - Cluster
+  - Pool
+  - Object store
```
Comment (kokhang) [inline Documentation/upgrade.md]: We also have Filesystem and Volumeattachment TPRs

File: Documentation/upgrade.md [hunk 2]
```diff
@@ -20,6 +20,10 @@ With this manual upgrade guide, there are a few notes to consider:
...
 * Rook is still in an alpha state.  Migrations and general support for breaking changes across versions are not supported or covered in this guide.
 * This guide assumes that your Rook operator and its agents are running in the `rook-system` namespace. It also assumes that your Rook cluster is in the `rook` namespace.  If any of these components is in a different namespace, search/replace all instances of `-n rook-system` and `-n rook` in this guide with `-n <your namespace>`.
+* Rook cluster installations on Kubernetes prior to version 1.7.x, use [ThirdPartyResource](https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-third-party-resource/) that have been deprecated as of 1.7 and removed in 1.8. When upgrading your Kubernetes cluster Rook TPRs have to be migrated to CustomResourceDefinition (CRD) following [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/migrate-third-party-resource/). Rook TPRs that require migration during upgrade are:
```
Comment (kokhang) [inline Documentation/upgrade.md]: i feel this should be done in a different section. This section is about Rook upgrade. The case where the TPR migration is relevant is during a K8S upgrade.

Comment (kokhang) [inline Documentation/upgrade.md]: I feel this needs to go to a different section. This is about rook upgrade. The case where TPR migration is relevant is during a K8S upgrade.

File: Documentation/upgrade.md (hunk 1 above)
Comment (kokhang) [inline Documentation/upgrade.md]: There is also Filesystem and VolumeAttachment TPRs. Also it is ObjectStore TPR.

File: Documentation/upgrade.md (hunk 2 above)
Comment (paha) [reply Documentation/upgrade.md]: @kokhang I agree, but didn't see a more reasonable place, do you have a suggestion for a better location?

I didn't want to create a new documentation section or a page. The  *Considerations* section in Upgrade doc sounded like the best option.

**Comment by travisn** [comment]
@paha you could add `[skip ci]` to the PR

**Comment by travisn** [comment]
what happens if someone has a Rook cluster running on 1.6, and they upgrade to 1.7 or 1.8? Our cluster is expected to continue running, right? All the resources (secrets, configmaps, services, deployments, pods, etc) should be fine. Just to be clear, what breaks?

File: Documentation/upgrade.md (hunk 2 above)
Comment (kokhang) [reply Documentation/upgrade.md]: How about creating a new sub heading under this same doc. Title it "Upgrade Kubernetes" and make it same level as "Upgrade Rook Software"

File: Documentation/upgrade.md [hunk 3]
```diff
@@ -314,3 +314,12 @@ kubectl -n rook get pod -l app=rook-ceph-mds -o jsonpath='{range .items[*]}{.met
 
 ## Completion
 At this point, your Rook cluster should be fully upgraded to running version `rook/rook:v0.5.1` and the cluster should be healthy according to the steps in the [health verification section](#health-verification).
+
+## Upgrading Kubernetes cluster
```
Comment (travisn) [inline Documentation/upgrade.md]: what about a title of `Upgrading Kubernetes`

File: Documentation/upgrade.md [hunk 4]
```diff
@@ -314,3 +314,12 @@ kubectl -n rook get pod -l app=rook-ceph-mds -o jsonpath='{range .items[*]}{.met
 
 ## Completion
 At this point, your Rook cluster should be fully upgraded to running version `rook/rook:v0.5.1` and the cluster should be healthy according to the steps in the [health verification section](#health-verification).
+
+## Upgrading Kubernetes
+:warning:
```
Comment (kokhang) [inline Documentation/upgrade.md]: This warning logo appears on its own line. Perhaps we should add a sentence beside it. Can you add that this is only relevant if we were originally using K8s 1.6 or older? If the user is migrating from 1.7 to 1.8, this is not relevant. Perhaps a one-sentence title should make it more clear

File: Documentation/upgrade.md [hunk 5]
```diff
@@ -314,3 +314,12 @@ kubectl -n rook get pod -l app=rook-ceph-mds -o jsonpath='{range .items[*]}{.met
 
 ## Completion
 At this point, your Rook cluster should be fully upgraded to running version `rook/rook:v0.5.1` and the cluster should be healthy according to the steps in the [health verification section](#health-verification).
+
+## Upgrading Kubernetes
+:warning:  
+Rook cluster installations on Kubernetes prior to version 1.7.x, use [ThirdPartyResource](https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-third-party-resource/) that have been deprecated as of 1.7 and removed in 1.8. When upgrading your Kubernetes cluster Rook TPRs have to be migrated to CustomResourceDefinition (CRD) following [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/migrate-third-party-resource/). Rook TPRs that require migration during upgrade are:
```
Comment (kokhang) [inline Documentation/upgrade.md]: I will add this ` If upgrading your Kubernetes cluster from 1.6.x or older, Rook TPRs have to be migrated...`

**Pull request merged**